modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/FileEntryFileEntryItemSelectorReturnTypeResolver.java

```diff
@@ -79,12 +86,30 @@ public String getValue(FileEntry fileEntry, ThemeDisplay themeDisplay)
 			"url", previewURL
 		).put(
 			"uuid", fileEntry.getUuid()
-		).toString();
+		);
+
+		if (ArrayUtil.contains(
+				PropsValues.DL_FILE_ENTRY_PREVIEW_VIDEO_MIME_TYPES,
+				fileEntry.getMimeType()) ||
+			Objects.equals(
+				DLContentTypes.VIDEO_EXTERNAL_SHORTCUT,
+				fileEntry.getMimeType())) {
+
+			jsonObject.put(
+				"html",
+				_dlVideoRenderer.renderHTML(
+					fileEntry.getFileVersion(), themeDisplay.getRequest()));
```

Comment:
You're using the video renderer for both images and videos. Is that on purpose?

Reply:
video render only if mimeType is DL_FILE_ENTRY_PREVIEW_VIDEO_MIME_TYPES or VIDEO_EXTERNAL_SHORTCUT

no images.